File: Bridge/System/IO/TextReader.cs
using Bridge;

namespace System.IO
{
    public abstract class TextReader : IDisposable {
        sealed class NullTextReader : TextReader
        {
            public override string ReadLine()
            {
                throw new NotImplementedException();
            }

            public override string ReadToEnd()
            {
                throw new NotImplementedException();
            }
        }

        public static readonly TextReader Null = new NullTextReader();

        protected TextReader()
        {
        }

        public virtual void Close()
        {
            throw new NotImplementedException();
        }

        public void Dispose()
        {
            throw new NotImplementedException();
        }
        
        public virtual int Peek()
        {
            throw new NotImplementedException();
        }

        public virtual int Read()
        {
            throw new NotImplementedException();
        }

        public virtual int Read(char[] buffer, int index, int count)
        {
            throw new NotImplementedException();
        }

        public virtual int ReadBlock(char[] buffer, int index, int count)
        {
            throw new NotImplementedException();
        }

        public virtual string ReadLine()
        {
            throw new NotImplementedException();
        }

        public virtual string ReadToEnd()
        {
            throw new NotImplementedException();
        }

        public static TextReader Synchronized(TextReader reader)
        {
            throw new NotImplementedException();
        }
    }
}

File: Bridge/Resources/.generated/system/iO/textReader.js
    Bridge.define("System.IO.TextReader", {
        inherits: [System.IDisposable],
        statics: {
            fields: {
                null: null
            },
            ctors: {
                init: function () {
                    Object.defineProperty(this, "null", {
                        get: function() {
                            return this._null$ === undefined ? (this._null$ = new System.IO.TextReader.NullTextReader()) : this._null$;
                        },
                        set: function(v) {
                            this._null$ = v;
                        }
                    });
                }
            },
            methods: {
                synchronized: function (reader) {
                    throw new System.NotImplementedException();
                }
            }
        },
        alias: ["dispose", "System$IDisposable$dispose"],
        ctors: {
            ctor: function () {
                this.$initialize();
            }
        },
        methods: {
            close: function () {
                throw new System.NotImplementedException();
            },
            dispose: function () {
                throw new System.NotImplementedException();
            },
            peek: function () {
                throw new System.NotImplementedException();
            },
            read: function () {
                throw new System.NotImplementedException();
            },
            read$1: function (buffer, index, count) {
                throw new System.NotImplementedException();
            },
            readBlock: function (buffer, index, count) {
                throw new System.NotImplementedException();
            },
            readLine: function () {
                throw new System.NotImplementedException();
            },
            readToEnd: function () {
                throw new System.NotImplementedException();
            }
        }
    });
